File: java/src/main/java/ch4_quicksort/SumArray.java
package ch4_quicksort;

public class SumArray {
    public static void main(String[] args) {
        int[] a = {1, 2, 3, 4};
        System.out.println(sumArray(a));
    }

    public static int sumArray(int[] arr) {
        return sumArray(arr, 0);
    }

    private static int sumArray(int[] arr, int index) {
        if (index == arr.length - 1) {
            return arr[index];
        }
        return arr[index] + sumArray(arr, index + 1);
    }
}

File: python/04_quicksort/sum_array.py
def sum_array(arr, index=0):
    if len(arr) == 0:
        return 0
    elif index == len(arr) - 1:
        return arr[index]
    else:
        return arr[index] + sum_array(arr, index + 1)


if __name__ == "__main__":
    print(sum_array([4, 5, 10]))
